lib/propmangle.js

```diff
@@ -202,9 +207,21 @@ function mangle_properties(ast, options) {
 
         var mangled = cache.props.get(name);
         if (!mangled) {
-            do {
-                mangled = base54(++cache.cname);
-            } while (!can_mangle(mangled));
+            if (debug) {
+                // debug mode: use a prefix and suffix to preserve readability, e.g. o.foo -> o._$foo$NNN_.
+                var debug_mangled = "_$" + name + "$" + debug_cache_name + "_";
+                
+                if (can_mangle(debug_mangled))
+                    mangled = debug_mangled;
+                else
+                    mangled = name;
```

Comment:
Yes. I created such a hand-crafted collision with your last PR.

I recognize this is debug code, but the bar for correctness should not be lower.